x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+- **Bulk Actions** - `POST /rules/_bulk_action`:
+  This endpoint includes many types of actions, but only the **bulk edit** and the **duplicate** actions should perform a migration on write.
+
+This endpoint also includes a `dry_run` mode that is executed to evaluate preconditions and warn the user before executing the actual request. No migration logic should take place for dry run requests, i.e when `dry_run=true`, since we never write to ES when this parameter is set to `true`.
+
+For the **bulk edit** action, we can take advantage of the `ruleParamsModifier` to carry out the migration, regardless of the type of edit that is being performed. See implementation details in the below [Bulk editing rules](#bulk-editing-rules) section.
+
+For the **duplicate** rule action:
+
+Since we will be creating a new rule on ES, we should create it with the new schema. Per definition, all duplicated rules will be `custom` rules. That means that for all rules -including prebuilt rules-, when duplicating, the `rule_source` field will not be created on the newly created rule.
+This action will not perform a migration-on-write of the original rule being duplicated for two reasons:
+
+- it would violate the principle of least surprise for the endpoint
+- the current implementation of the endpoint does not modify the original rule. This gives us no window of opportunity to migrate the rule and save to ES without adding performance overhead.
+
+All other type of actions should **not** perform migration-on-write:
+
+- Enable
+- Disable
+- Delete
+- Export: see Importing and Exporting Rules section below
+
+### Technical implementation of migration-on-write
+
+The logic for the migration of the rule saved objects, which means the determination of the `immutable` and `rule_source` fields before writing to ES, might differ depending on the action being performed by the user.
+
+Let's see all possible use cases that will require migration-on-write, the endpoints that they apply to, and the expected resulting migrated field, based on the action and their input.
+
+#### Updating and upgrading rules
+
+Updating rules can currently be performed via five endpoints:
+
+- **Update Rule** - `PUT /rules`
+- **Patch Rule** - `PATCH /rules`
+- **Bulk Update Rules** - `PUT /rules/_bulk_update`
+- **Bulk Patch Rules** - `PATCH /rules/_bulk_update`
+- **Bulk Actions** - `POST /rules/_bulk_action`: with **bulk edit** action
+
+Upgrading prebuilt rules to their newer version is done by two endpoints:
+
+- **(LEGACY) Install Prebuilt Rules And Timelines** - `PUT /rules/prepackaged`
+- **Perform Rule Upgrade** - `POST /prebuilt_rules/upgrade/_perform` (Internal)
+
+The legacy endpoint does not allow for customization of fields during the upgrade, but the new rule upgrade customization endpoint does. 
+
+Additionally:
+
+- **Bulk Actions** - `POST /rules/_bulk_action`: with **duplicate** action
+
+will perform migration but does not allow for customization during the duplication proces.
```

Comment:
type `proces` -> `process`

Why should Bulk duplicate allow customization? Or the meaning of this sentence is to highlight that Bulk duplicate doesn't accept any params besides rule ids?